bootstrap/src/lib/cache/index.ts

```diff
@@ -150,19 +172,24 @@ export const withCache: Middleware<CacheOptions> = async (execute, options = def
       ? await _getWithCoalescing()
       : await cache.get(key)
 
+    const requestMaxAge = _getRequestMaxAge(request)
     if (entry) {
-      if (maxAge >= 0) {
+      if (!requestMaxAge || requestMaxAge >= 0) {
         logger.debug(`Cache: GET ${key}`, entry)
-        if (maxAge !== entry.maxAge) await _cacheOnSuccess(entry)
-        return { jobRunID: data.id, ...entry }
+        // If maxAge is present on the request, cache entry is updated
+        if (requestMaxAge && requestMaxAge !== entry.maxAge)
+          await _cacheOnSuccess(entry, requestMaxAge)
+        rateLimit.incrementParticipantHeartbeat(_getKey(request))
```

Comment:
If the request hit the cache, we still increment the participant heartbeat. We assume is a valid request, as it exists in the cache